cylc/flow/config.py

```diff
@@ -538,8 +542,6 @@ def __init__(
 
         self.mem_log("config.py: before load_graph()")
         self.load_graph()
-        if self._is_validate():
-            GraphNodeParser.get_inst().clear()
```

Comment:
Looking at the code nothing is done to `GraphNodeParser` until this line so wiping the instance will have no effect.

Reply:
Are you sure? As far as I can tell, there is this chain of calls:

1. `self.load_graph()` in the line above
2. In `load_graph()`: https://github.com/cylc/cylc-flow/blob/f656c97e5ac81144635d2bcb8bfb69b48d4f75ae/cylc/flow/config.py#L2215-L2216
3. In `_proc_triggers()`: https://github.com/cylc/cylc-flow/blob/f656c97e5ac81144635d2bcb8bfb69b48d4f75ae/cylc/flow/config.py#L2232
4. In `generate_task_defs()`: https://github.com/cylc/cylc-flow/blob/f656c97e5ac81144635d2bcb8bfb69b48d4f75ae/cylc/flow/config.py#L1638

Reply:
Yeah, GraphNodeParser caches already-parsed nodes. On master, this `clear()` results in **more** parsing during a validation, due to reparsing nodes that were already parsed.  So I'm not sure what that line is there for???